pkg/framework/plugins/removepodsviolatingnodetaints/node_taint_test.go

```diff
@@ -322,6 +323,42 @@ func TestDeletePodsViolatingNodeTaints(t *testing.T) {
 			expectedEvictedPodCount: 0, // p2 and p7 can't be evicted
 			nodeFit:                 true,
 		},
+		{
+			description:             "Pods tolerating included taints should not get evicted",
+			pods:                    []*v1.Pod{p1},
+			nodes:                   []*v1.Node{node1},
+			evictLocalStoragePods:   false,
+			evictSystemCriticalPods: false,
+			includedTaints:          []string{"testTaint1=test1"},
+			expectedEvictedPodCount: 0, // nothing gets evicted, as p1 tolerates the included taint
+		},
+		{
+			description:             "Pods not tolerating not included taints should not get evicted",
+			pods:                    []*v1.Pod{p1},
+			nodes:                   []*v1.Node{node1},
+			evictLocalStoragePods:   false,
+			evictSystemCriticalPods: false,
+			includedTaints:          []string{"testTaint2=test2"},
+			expectedEvictedPodCount: 0, // nothing gets evicted, as taint is not included.
+		},
+		{
+			description:             "Pods not tolerating included taints should get evicted",
+			pods:                    []*v1.Pod{p1, p2, p3},
+			nodes:                   []*v1.Node{node1},
+			evictLocalStoragePods:   false,
+			evictSystemCriticalPods: false,
+			includedTaints:          []string{"testTaint1=test1"},
+			expectedEvictedPodCount: 1, // p1 and p3 tolerate the included taint, p2 gets evicted
+		},
+		{
+			description:             "Pods not tolerating not included taints should not get evicted",
+			pods:                    []*v1.Pod{p1, p2, p4},
+			nodes:                   []*v1.Node{node1},
+			evictLocalStoragePods:   false,
+			evictSystemCriticalPods: false,
+			includedTaints:          []string{"testTaint1=test2"},
```

Comment:
This case tests the same case as "Pods not tolerating not included taints should not get evicted". I.e. a node is not tainted with `includedTaints` taints. So no toleration is needed. Worth adding a test where `includeTaints` is empty, a node is tainted with two taints, pods has a list of pods where some tolerate both taints, some don't.

Reply:
You're right. I merged both "Pods not tolerating not included taints should not get evicted" tests and created a new one "Pods not tolerating all taints are evicted when includedTaints is empty".